meta-mender-commercial/recipes-mender/mender-gateway/mender-gateway.inc

```diff
@@ -0,0 +1,49 @@
+LICENSE = "CLOSED"
+LICENSE_FLAGS = "commercial"
+
+inherit systemd
+
+SUB_FOLDER_arm = "arm"
+SUB_FOLDER_aarch64 = "aarch64"
+SUB_FOLDER_x86-64 = "x86_64"
+
+COMPATIBLE_HOSTS = "arm|aarch64|x86_64"
+
+FILES_${PN} = " \
+    ${bindir}/mender-gateway \
+"
+
+FILES_${PN}_append_mender-systemd = " \
+    ${systemd_system_unitdir}/mender-gateway.service \
+"
+
+S = "${WORKDIR}/mender-gateway-${PV}"
+
+do_version_check() {
+    if ! ${@'true' if d.getVar('MENDER_DEVMODE') else 'false'}; then
+        if ! strings ${WORKDIR}/${SUB_FOLDER}/mender-gateway | fgrep -q "${PN} ${PV}"; then
+            bbfatal "String '${PN} ${PV}' not found in binary. Is it the correct version? Check with --version. Possible candidates: $(strings ${WORKDIR}/${SUB_FOLDER}/mender-gateway | grep '${PN} [a-f0-9]')"
+        fi
+    fi
+}
+addtask do_version_check after do_unpack before do_install
+
+SYSTEMD_SERVICE_${PN}_mender-systemd = "mender-gateway.service"
+
+do_install() {
+    install -d -m 755 ${D}${bindir}
```

Comment:
I think that is unnecessary?

Reply:
Maybe unnecessary, but is it harmless? I think it is a good practice to install the dir before the files. We do it in most of our recipes.

Reply:
Certainly not harmful, just an observation.

Reply:
AFAIK, this is necessary in all recipes. Directories in `${D}` do not generally pre-exist.

Even if there might be an exception for `bindir` (I haven't checked) I think we should keep it for consistency.